### PR TITLE
predicates/traffic: fix flaky TrafficSegment tests

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Test failure could be reproduced via:
```
go test ./predicates/traffic/ -count=100 -run=TestTrafficSegmentSplit
```

* use delta of expected value instead of total number of test requests
* use predefined number of test requests for all tests

Based on the biggest difference seen for 1000 test requests:
```
Max difference between 500 and 462 allowed is 25, but difference was 38
```
the delta should be 38/500=0.076

To reduce flakiness:

* increase total number of samples to 10'000
* then increase delta to 6% based on the failed test case output

### CODE REVIEW

File: predicates/traffic/segment_test.go [hunk 1]
```diff
@@ -51,20 +51,30 @@ func requestWithR(r float64) *http.Request {
 	return req
 }
 
-func doN(t *testing.T, n int, client *proxytest.TestClient, request func() *http.Request) map[int]int {
-	codes := make(map[int]int)
+// doN performs a number of requests and returns the number of responses for each status code and
+// a total number of requests performed.
+// Results use float64 type to simplify fractional comparisons.
+func doN(t *testing.T, client *proxytest.TestClient, request func() *http.Request) (map[int]float64, float64) {
```
Comment (szuecs) [inline predicates/traffic/segment_test.go:57]: Where is the N?
Maybe better to call it `do` ?

**Comment by szuecs** [comment]
:+1:

File: predicates/traffic/segment_test.go (hunk 1 above)
Comment (AlexanderYastrebov) [reply predicates/traffic/segment_test.go:57]: it is to highlight that it does multiple requests

**Comment by AlexanderYastrebov** [comment]
:+1:

File: predicates/traffic/segment_test.go [hunk 2]
```diff
@@ -129,18 +139,13 @@ func TestTrafficSegmentSplit(t *testing.T) {
...
-		N     = 1_000
-		delta = 0.05 * N
-	)
-
-	codes := getN(t, N, p.Client(), p.URL+"/test")
+	codes, n := getN(t, p.Client(), p.URL+"/test")
 
 	t.Logf("Response codes: %v", codes)
 
-	assert.InDelta(t, N*0.5, codes[200], delta)
```
Comment (MustafaSaber) [inline predicates/traffic/segment_test.go:141]: Why moving to a function?

**Pull request merged**